### ISSUE TITLE
can't open the right header file

### ISSUE THREAD

**Issue by caurier** [microsoft/vscode-cpptools#1253]
When `#include <sys/socket.h>`, then `Goto Definition`, `/usr/include/asm-generic/socket.h` is opened.
However, the right header file should be `/usr/include/x86_64-linux-gnu/sys/socket.h`.

follow is my "c_cpp_properties.json"
```
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "/usr/include/c++/5",
                "/usr/include/x86_64-linux-gnu/c++/5",
                "/usr/include/c++/5/backward",
                "/usr/lib/gcc/x86_64-linux-gnu/5/include",
                "/usr/local/include",
                "/usr/lib/gcc/x86_64-linux-gnu/5/include-fixed",
                "/usr/include/x86_64-linux-gnu",
                "/usr/include",
                "${workspaceRoot}/common"
            ],
            "intelliSenseMode": "clang-x64",
            "browse": {
                "path": [
                    "/usr/include/c++/5",
                    "/usr/include/x86_64-linux-gnu/c++/5",
                    "/usr/include/c++/5/backward",
                    "/usr/lib/gcc/x86_64-linux-gnu/5/include",
                    "/usr/local/include",
                    "/usr/lib/gcc/x86_64-linux-gnu/5/include-fixed",
                    "/usr/include/x86_64-linux-gnu",
                    "/usr/include",
                    "${workspaceRoot}",
                    "${workspaceRoot}/common"
                ]
            }
        }
    ],
    "version": 3
}
```

**Comment by sean-mcmanus** [comment]
Does the correct file appear in the list of available options on the right when you use Peek Definition? We plan to fix the issue with the sys part of the #include not being taken into account to filter out the invalid targets.

**Comment by caurier** [comment]
Peek Definition is the same.

**Comment by sean-mcmanus** [comment]
This should be fixed in our next release (the full path of the include is used to filter out the invalid file targets).

**Comment by sean-mcmanus** [comment]
@caurier Fixed with 0.17.4.
